plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/procedure/TestIcebergAddFilesProcedure.java

```diff
@@ -0,0 +1,556 @@
+    {
+        String hiveTableName = "test_migrate_recursive_directory_" + randomNameSuffix();
+        String icebergTableName = "test_migrate_recursive_directory_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE hive.tpch." + hiveTableName + " WITH (format = 'ORC') AS SELECT 1 x", 1);
+        assertUpdate("CREATE TABLE iceberg.tpch." + icebergTableName + "(x int)");
+
+        // Move a file to the nested directory
+        String path = (String) computeScalar("SELECT \"$path\" FROM hive.tpch." + hiveTableName);
+        String fileName = Location.of(path).fileName();
+        Path tableLocation = dataDirectory.resolve("tpch").resolve(hiveTableName);
+        Files.createDirectory(tableLocation.resolve("nested"));
+        Files.move(tableLocation.resolve(fileName), tableLocation.resolve("nested").resolve(fileName));
+
+        assertQueryFails(
+                "CALL iceberg.system.add_files(target_schema_name=>'tpch', target_table_name=>'" + icebergTableName + "', location=>'" + tableLocation + "', format=>'ORC')",
+                ".*Recursive directory must not exist when recursive_directory argument is 'fail'.*");
+        assertQueryFails(
+                "CALL iceberg.system.add_files(target_schema_name=>'tpch', target_table_name=>'" + icebergTableName + "', location=>'" + tableLocation + "', format=>'ORC', recursive_directory=>'fail')",
+                ".*Recursive directory must not exist when recursive_directory argument is 'fail'.*");
+
+        assertUpdate("CALL iceberg.system.add_files(target_schema_name=>'tpch', target_table_name=>'" + icebergTableName + "', location=>'" + tableLocation + "', format=>'ORC', recursive_directory=>'false')");
+        assertQueryReturnsEmptyResult("SELECT * FROM " + icebergTableName);
+
+        assertUpdate("CALL iceberg.system.add_files(target_schema_name=>'tpch', target_table_name=>'" + icebergTableName + "', location=>'" + tableLocation + "', format=>'ORC', recursive_directory=>'true')");
+        assertQuery("SELECT * FROM " + icebergTableName, "VALUES 1");
+
+        assertUpdate("DROP TABLE hive.tpch." + hiveTableName);
+        assertUpdate("DROP TABLE iceberg.tpch." + icebergTableName);
+    }
+
+    @Test
+    void testAddFilesDirectoryLocation()
+    {
+        String hiveTableName = "test_add_files_location_" + randomNameSuffix();
+        String icebergTableName = "test_add_files_location_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE iceberg.tpch." + icebergTableName + " AS SELECT 1 x", 1);
+        assertUpdate("CREATE TABLE hive.tpch." + hiveTableName + " AS SELECT 2 x", 1);
+
+        String path = (String) computeScalar("SELECT \"$path\" FROM hive.tpch." + hiveTableName);
+        String directory = Location.of(path).parentDirectory().toString();
+
+        assertUpdate("CALL iceberg.system.add_files(target_schema_name=>'tpch', target_table_name=>'" + icebergTableName + "', location=>'" + directory + "', format=>'ORC')");
```

Comment:
if my target iceberg table is partitioned, in which partition am i adding the files from the directory?

Reply:
It's not supported as `TestIcebergAddFilesProcedure.testAddFilesToPartitionTableWithLocation`. I will add the caveat to docs.